### PR TITLE
feature extraction

### PR DESCRIPTION
**Pull request by tvannoy**
Feature extraction functions for our latest-and-greatest features!

### CODE REVIEW

File: feature_testing.m [hunk 1]
```diff
@@ -0,0 +1,87 @@
+%% Kyle's Dumb Tests
```
Comment (rustky) [inline feature_testing.m:1]: Probably not needed haha

Comment (tvannoy) [reply feature_testing.m:1]: I'd be fine with getting rid of this whole file.

File: allDataCreation.m [hunk 2]
```diff
@@ -19,10 +19,10 @@
 % Iterate through all of the files. Throw out non-insect-image
 % files/folders
 for i = 1:length(days)
-    temp_folders = {dir(strjoin([basefilepath days{i}],'\\')).name};
+    temp_folders = {dir(string([basefilepath '\' days{i}])).name};
```
Comment (tvannoy) [inline allDataCreation.m:22]: Make this platform-independent:
```matlab
temp_folders = {dir(string([basefilepath filesep days{i}])).name};
```

File: getData.m [hunk 3]
```diff
@@ -1,8 +1,8 @@
 function data = getData(basefilepath,day,folder,image)
-    adjusted_data_decembercal = load([basefilepath day '\' folder '\' 'adjusted_data_decembercal.mat']);
+    load([basefilepath '\' day '\' folder '\' 'adjusted_data_decembercal.mat']);
```
Comment (tvannoy) [inline getData.m:2]: Same comment as before. Use `filesep` so this works on all operating systems

**Comment by rustky** [comment]
Noted and fixed

**Comment by tvannoy** [comment]
@rustky If your code is done and working, I can merge this in.

**Comment by rustky** [comment]
still need to test. I'll let you know when we're good

**Comment by tvannoy** [comment]
@tscofield034 The relevant functions that @rustky wrote are `allDataCreation.m` and potentially `getData.m`. I'm not sure how to use the functions, but there's a short example in `allocateTestTrainData.m`. 

The rest of functions are mostly related to feature extraction. I'll merge this in later today.

**Comment by rustky** [comment]
@tvannoy @tscofield034  The `allDataCreation.m` should have some comments describing what it does so hopefully it won't be too bad to use. It is more or less a one-time run to create the large set of all files. The `getData.m` is a function operating within a for loop that takes in a line from the large set of all files and grabs its normalized data. At the point of its calling the randomization should have taken place, so you already know which lines of the large dataset you intend to grab. Hope this clears some stuff up. The first week of my internship is a little slow, so don't hesitate to reach out if you have any questions.

**Comment by tvannoy** [comment]
@rustky What do the `dayStart` and `dayEnd` integer arguments in `allDataCreation` mean?

**Comment by rustky** [comment]
@tvannoy indexes the primary folder to grab the folders with the expected file structure

**Comment by tvannoy** [comment]
> @tvannoy indexes the primary folder to grab the folders with the expected file structure

@rustky :+1: gotcha.

**Pull request merged**